client/header/activity-panel/unread-indicators.js

```diff
@@ -57,3 +72,71 @@ export function getUnreadNotes( select ) {
 export function getLowStockCount() {
 	return getSetting( 'lowStockCount', 0 );
 }
+
+function getIncompleteTasksCount( tasks, dismissedTasks ) {
+	if ( ! tasks ) {
+		return 0;
+	}
+	return tasks.filter(
+		( task ) =>
+			task.visible &&
+			! task.completed &&
+			! dismissedTasks.includes( task.key )
+	).length;
+}
+
+function getAbbreviatedNotifications( select ) {
+	const { getOption } = select( OPTIONS_STORE_NAME );
+	const { getNotes } = select( NOTES_STORE_NAME );
+	const storeAlerts = getNotes( ALERTS_QUERY );
+	const thingsToDoNext = applyFilters(
+		'woocommerce_admin_onboarding_task_list',
+		[]
+	);
+	const dismissedTasks = getOption( 'woocommerce_task_list_dismissed_tasks' );
+	const storeAlertsCount = storeAlerts.length ?? 0;
+	const orderStatuses = getOrderStatuses( select );
+	const notifications = [
+		{
+			name: 'thingsToDoNext',
+			count:
+				getIncompleteTasksCount( thingsToDoNext, dismissedTasks ) +
+				storeAlertsCount,
+			critical: storeAlertsCount,
+		},
+		{
+			name: 'ordersToProcess',
+			count: getUnreadOrders( select, orderStatuses ),
+		},
+		{
+			name: 'reviewsToModerate',
+			count: getUnapprovedReviews( select ),
+		},
+		{
+			name: 'stockNotices',
+			count: getLowStockProducts( select ),
+		},
+	];
+
+	return applyFilters(
+		'woocommerce_admin_abbreviated_notifications',
```

Comment:
I seem to have gotten the idea in my head that we didn't want to continue using the pattern of "filtered array of data that turns into components". Should we be using SlotFill here instead? cc: @samueljseay _(note: yes, I continued the components-as-data pattern [here](https://github.com/Automattic/woocommerce-payments/pull/1799/files#diff-062eb1e5dc64b4146f2536ba4987039bedb128b2d938de7217119a9beaa2fd87R10), but those components need to slot in to our existing task list filter of data-represented-components)_ 😅

Reply:
@octaedro sorry if the messaging hasn't been too clear on this, but yeah even aside from "filtered array of data that turns into components" not being a great pattern there are too many gotchas with JS filters that make them really hard to use if you're trying to use them as an extensibility point. Since that really only leaves us with SlotFill as an alternative I think we need to go that route here. 

I realise that SlotFill can take some time to get your head around so please do reach out to @jeffstieler, @louwie17 or even @psealock or anyone from Gezora because they've all had experience implementing this pattern.

Reply:
Perhaps the easiest path forward would be to remove the extensibility from this PR and work on it in a follow up?

Reply:
Thank you @jeffstieler and @samueljseay, you're right, using SlotFill for the abbreviated panels is a better idea. If it's ok with you I'll remove the extensibility and work on a SlotFill approach in a follow-up PR.
The extensibility was removed in the commit fbb3aaa